Platforms/QemuQ35Pkg/PlatformBuild.py

```diff
@@ -314,68 +314,51 @@ def FlashRomImage(self):
             VirtualDrive = VirtualDriveManager(VirtualDrivePath, self.env)
-            self.env.SetValue("VIRTUAL_DRIVE_PATH", VirtualDrivePath, "Set Virtual Drive path in case not set")
-            ut = UnitTestSupport(os.path.join(output_base, "X64"))
-
-            if empty_drive and os.path.isfile(VirtualDrivePath):
-                    os.remove(VirtualDrivePath)
-
-            if not os.path.isfile(VirtualDrivePath):
-                VirtualDrive.MakeDrive()
-
-            test_regex = self.env.GetValue("TEST_REGEX", "")
-
-            if test_regex != "":
-                ut.set_test_regex(test_regex)
-                ut.find_tests()
-                ut.copy_tests_to_virtual_drive(VirtualDrive)
-
-            if run_tests:
-                if test_regex == "":
-                    logging.warning("No tests specified using TEST_REGEX flag but RUN_TESTS is TRUE")
-                elif not empty_drive:
-                    logging.info("EMPTY_DRIVE=FALSE. This could impact your test results")
-
-                if not shutdown_after_run:
-                    logging.info("SHUTDOWN_AFTER_RUN=FALSE (default). XML test results will not be \
-                        displayed until after the QEMU instance ends")
-                ut.write_tests_to_startup_nsh(startup_nsh)
-
-            nshpath = os.path.join(output_base, "startup.nsh")
-            startup_nsh.WriteOut(nshpath, shutdown_after_run)
+        else:
+            VirtualDrivePath = self.env.GetValue("VIRTUAL_DRIVE_PATH", os.path.join(output_base, "VirtualDrive.img"))
+            VirtualDrive = LinuxVirtualDriveManager (VirtualDrivePath)
 
-            VirtualDrive.AddFile(nshpath)
+        self.env.SetValue("VIRTUAL_DRIVE_PATH", VirtualDrivePath, "Set Virtual Drive path in case not set")
 
-        else:
-            VirtualDrivePath = self.env.GetValue("VIRTUAL_DRIVE_PATH", os.path.join(output_base, "VirtualDrive"))
-            logging.warning("Linux currently isn't supported for the virtual drive. Falling back to an older method")
+        if empty_drive and os.path.isfile(VirtualDrivePath):
+            os.remove(VirtualDrivePath)
 
-            test_regex = self.env.GetValue("TEST_REGEX", "")
+        if not os.path.isfile(VirtualDrivePath):
+            VirtualDrive.MakeDrive()
 
-            if run_tests:
-                logging.critical("Linux doesn't support running unit tests due to lack of VHD support")
+        ut = UnitTestSupport(os.path.join(output_base, "X64"))
+        if test_regex != "":
+            ut.set_test_regex(test_regex)
+            ut.find_tests()
+            ut.copy_tests_to_virtual_drive(VirtualDrive)
 
-            if os.path.exists(VirtualDrivePath) and empty_drive:
-                shutil.rmtree(VirtualDrivePath)
+        if run_tests:
+            if test_regex == "":
+                logging.warning("No tests specified using TEST_REGEX flag but RUN_TESTS is TRUE")

```

Comment:
I think default behavior would be that TEST_REGEX is set to run all tests if RUN_TESTS is true. It seems silly that someone that wants to run all tests would need to do RUN_TESTS=TRUE TEST_REGEX=**/* (or whatever the proper regex is)

If that is the default behavior by the regex being "", then I don't think you need to throw a warning at the user. This would be expected behavior in my opinion

Reply:
If that is the default behavior by the regex being "", then I don't think you need to throw a warning at the user. This would be expected behavior in my opinion

Reply:
I'm not sure I understand.

TEST_REGEX defines what tests are copied into the virtual drive and RUN_TESTS causes the tests to be added to the startup.nsh script for automated execution. There are cases where I only want to copy tests over and not have them run automatically but never a case where I want to run tests automatically but didn't actually copy any over. There's also no way to define what a test is other than by filename and module type (UEFI_APPLICATION).

Reply:
Understood. I didn't think about the fact that you may want to bring tests over but not run them. I also thought had a way of knowing what tests we were compiling, and the test regex was just filtering on that test list.